src/tree2d.cpp

```diff
@@ -37,14 +37,16 @@ namespace manifold {
 // Recursive sorting is not the most efficient, but simple and guaranteed to
 // result in a balanced tree.
 void BuildTwoDTreeImpl(VecView<PolyVert> points, bool sortX) {
-  using CmpFn = std::function<bool(const PolyVert&, const PolyVert&)>;
-  CmpFn cmpx = [](const PolyVert& a, const PolyVert& b) {
+  auto cmpx = [](const PolyVert& a, const PolyVert& b) {
     return a.pos.x < b.pos.x;
   };
-  CmpFn cmpy = [](const PolyVert& a, const PolyVert& b) {
+  auto cmpy = [](const PolyVert& a, const PolyVert& b) {
     return a.pos.y < b.pos.y;
   };
-  manifold::stable_sort(points.begin(), points.end(), sortX ? cmpx : cmpy);
+  if (sortX)
+    manifold::stable_sort(points.begin(), points.end(), cmpx);
+  else
+    manifold::stable_sort(points.begin(), points.end(), cmpy);
```

Comment:
Is this different than before?

Reply:
they are the same

Reply:
In that case, I think I like the ternary better - a bit cleaner.

Reply:
Oh sorry, by the same I mean the behavior. Performance is quite different.